src/gates.cpp

```diff
@@ -283,10 +283,12 @@ void doors::close_door( map &m, Creature &who, const tripoint &closep )
             if( !veh->handle_potential_theft( get_avatar() ) ) {
                 return;
             }
-            veh->close( closable );
-            //~ %1$s - vehicle name, %2$s - part name
-            who.add_msg_if_player( _( "You close the %1$s's %2$s." ), veh->name, veh->part( closable ).name() );
-            didit = true;
+            if( veh->can_close( closable ) ) {
```

Comment:
You did not pass `who` here.
```suggestion
            if( veh->can_close( closable, who ) ) {
```